### PR TITLE
[MKR-3] Prepare Asset Swapper for RFQM

### PR DESCRIPTION
**Pull request by phil-ociraptor**
## Description

Contains a small amount of refactoring to make our existing code support both RFQT and RFQM flows. Unit tests have been updated and new coverage is added for RFQM flows.

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

**Review by alexkroeger — APPROVED** [review]
lgtm, please fix static tests

File: packages/asset-swapper/src/utils/quote_requestor.ts [hunk 1]
```diff
@@ -236,42 +238,15 @@ export class QuoteRequestor {
...
             takerToken,
             assetFillAmount,
             marketOperation,
             comparisonPrice,
             _opts,
-            'indicative',
+            this._rfqtAssetOfferings,
         );
-
-        // validate
```
Comment (alexkroeger) [inline packages/asset-swapper/src/utils/quote_requestor.ts:249]: good to separate out this logic, consider a `validateQuote` function (non-async)

Comment (phil-ociraptor) [reply packages/asset-swapper/src/utils/quote_requestor.ts:249]: This is a bit trickier than I originally thought. It's straightforward to create two functions: `validateFirmQuote` and a `validateIndicativeQuote` - but gets a bit gnarly when trying to unify them into a single `validateQuote` fn. 

Thoughts on having two separate functions?

Comment (phil-ociraptor) [reply packages/asset-swapper/src/utils/quote_requestor.ts:249]: Started down the unified path and got here: https://github.com/0xProject/protocol/commit/0255b4deb2cff4d608bab9df4e85666167da776d

But haven't found something that I'm happy with. Will look into this in a separate PR if I have time

**Pull request merged**